workshop/6 - Operational Excellence/README.md

```diff
@@ -0,0 +1,192 @@
+    - Enter the following data:
+
+    |Name|Value|
+    |-----|-----|
+    |*Key*|Api:App:RetryDemo|
+    |*Value*|2|
+
+    ![Screenshot of Resource Group](../images/6-Operational%20Excellence/Monitoring-App-Config.png)
+
+1. Press **Apply**.
+1. Restart the API web app App Service
+    - Go to the API web app App Service (In your resource group resources, search for the app service that starts with `api-` and select the result that is of type **App Service**)
+
+        ![Screenshot of Resource Group](../images/6-Operational%20Excellence/Monitoring-App-Service.png)
+
+    - Navigate to the "Overview" blade
+    - Click the "Restart" button at the top of the page.
+  
+    > It will take a few minutes for the App Service to restart. When it restarts, the application will use the `Api:App:RetryDemo` configuration. You need to restart the App Service any time you update a configuration value. When the value of `Api:App:RetryDemo` is 2, the first request to the application API generates a 503 error. But the retry pattern sends a second request that is successful and generates a 200 response. We recommend using the Application Insights Live Metrics features to view the HTTP responses in near real-time.
+    > App Insights can up to a minute to aggregate the data it receives, and failed requests might not appear right away in the Failures view.
+
+1. To see the **Retry Pattern** in action you can click throughout the Relecloud website and should not see any impact to the user's ability to purchase a concert ticket. However, in App Insights you should see the 503 error happens for 50% of the requests sent to the Web API.
+
+1. Search for **Application Insights** from your resources and navigate to the **Application Maps** section. Application Map helps you spot performance bottlenecks or failure hotspots across all components of your distributed application. Each node on the map represents an application component or its dependencies and has health KPI and alerts status. You can select any component to get more detailed diagnostics, select the failing calls, and you will be able to dig through the details.
+
+    ![Screenshot of Resource Group](../images/6-Operational%20Excellence/Monitoring-App-Failures.png)
+
+1. On your right side of the screen, select **Investigate Failures** and select a transaction to go through End-to-End transaction details.
+
+    ![Screenshot of Resource Group](../images/6-Operational%20Excellence/Monitoring-App-503.png)
+
+> We recommend you cleanup by deleting the `Api:App:RetryDemo` setting.
+> Please note that this module simulates an error within the code, which is not recommended in a production environment. In real-world scenarios, failures may arise when connecting to other services or third-party APIs. Application Insights provides a comprehensive overview of such interim failures while your retry code ensures your app continues to function effectively. 
+
+#### Use Application Insights to gather custom telemetry
```

Comment:
this section is fine for now, but I think with an upgrade to .net 8 we should switch to open telemetry. that is supported by app insights and can use more .net primitives